builtin/logical/transit/backend.go

```diff
@@ -6,36 +6,46 @@ import (
 )
 
 func Factory(conf *logical.BackendConfig) (logical.Backend, error) {
-	return Backend().Setup(conf)
+	b := Backend()
+	be, err := b.Backend.Setup(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.policies.loadStoredPolicies(conf.StorageView)
+	if err != nil {
+		return nil, err
+	}
+
+	return be, nil
 }
 
-func Backend() *framework.Backend {
+func Backend() *backend {
 	var b backend
 	b.Backend = &framework.Backend{
-		PathsSpecial: &logical.Paths{
-			Root: []string{
-				"keys/*",
-			},
-		},
-
 		Paths: []*framework.Path{
 			// Rotate/Config needs to come before Keys
 			// as the handler is greedy
-			pathConfig(),
-			pathRotate(),
-			pathRewrap(),
-			pathKeys(),
-			pathEncrypt(),
-			pathDecrypt(),
-			pathDatakey(),
+			b.pathConfig(),
+			b.pathRotate(),
+			b.pathRewrap(),
+			b.pathKeys(),
+			b.pathEncrypt(),
+			b.pathDecrypt(),
+			b.pathDatakey(),
 		},
 
 		Secrets: []*framework.Secret{},
 	}
 
-	return b.Backend
+	b.policies = &policyCache{
+		cache: map[string]*lockingPolicy{},
+	}
+
+	return &b
 }
 
 type backend struct {
 	*framework.Backend
+	policies *policyCache
```

Comment:
You might as well embed policyCache instead of a pointer to one, to avoid one level of indirection